Fertility model (#710)
ihmeuw/vivarium_research: docs/source/other_models/pregnancy/index.rst

  decorator is encountered below. If you need an even deeper section level, just
  choose a new decorator symbol from the list here:
  https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#sections
  And then add it to the list of decorators above.

=========================
Pregnancy
=========================

.. contents::
   :local:
   :depth: 1

.. list-table:: Abbreviations
  :widths: 15 15 15
  :header-rows: 1

  * - Abbreviation
    - Definition
    - Note
  * - ASFR
    - Age-specific fertility rate
    - 
  * - SBR
    - Stillbirth to live birth ratio
    - 
  * - LBWSG
    - Low birthweight and short gestation
    - 

Overview
-------------

.. warning::

   The current version of this document details a strategy to model pregnancy in a simulation with a population of women of reproductive age only and does not yet detail a strategy to pair maternal child dyads.

   Such a strategy will need to distinguish bewteen stillbirths and live births and assign the child's LBWSG exposure category in a way that is consistent with the mother's duration of pregnancy.

GBD Modeling Strategy
----------------------

> Pregnancy and births are not explit outcomes in the GBD study. However, there are location- and year-specfic GBD covariates related to fertility and births, including:

Reviewer: ```suggestion
Pregnancy and births are not explicit outcomes in the GBD study. However, there are location- and year-specfic GBD covariates related to fertility and births, including:
```

.. list-table:: Covariates
  :widths: 15 15 15
  :header-rows: 1

  * - Covariate name
    - Covariate ID
    - Note
  * - Live births (thousands)
    - 60
    - 
  * - Live births by sex
    - 1106
    - 
  * - Live births by sex and maternal age
    - 2298
    - 
  * - Stillbirth to live birth ratio
    - 2267
    - Not specific to maternal age. Upper and lower bound estimates are equal to mean estimate (no uncertainty interval)
  * - Age-specific fertility rate
    - 13
    - 
  * - Total fertility rate
    - 2363
    - 

> GBD has estimated the prevalence of pregnancy (as in itermediate variable for the estimation of various outcomes) as:

Reviewer: ```suggestion
GBD has estimated the prevalence of pregnancy (as an intermediate variable for the estimation of various outcomes) as:
```

.. math::

   (ASFR + (SBR * ASFR)) * 46/52

Where :math:`ASFR` is the age-specific fertility rate, :math:`SBR` is the stillbirth to live birth ratio, and :math:`46/52` is the proportion of the year spent pregnant (40 weeks) and postpartum (6 weeks) [GBD-2019-Risk-Factors-Appendix-pregnancy]_.

.. todo::

>    Determine the treshold of gestational age at which a loss of pregnancy is classified as a stillbirth rather than miscarriage for the GBD covariate. Standard thresholds are 20 or 24 weeks.

Reviewer: ```suggestion
   Determine the threshold of gestational age at which a loss of pregnancy is classified as a stillbirth rather than miscarriage for the GBD covariate. Standard thresholds are 20 or 24 weeks.
```

Vivarium Modeling Strategy
----------------------------

We will model pregnancy as a characteristic of women of reproductive age in our simulations. We will inform the *incidence* of pregnancy using the age-specific fertility and stillbirth to live birth ratio covariates from GBD. We will inform the *duration* of pregnancy using the GBD 2019 exposure distribution of gestational age.

.. image:: diagram.svg

.. list-table:: State definitions
  :widths: 15 15 15
  :header-rows: 1

  * - State
    - Description
    - Note
  * - np
    - Not pregnant or postpartum
    - 
  * - p
    - Pregnant
    - 
  * - pp
    - Postpartum
    - 

.. note::

   We will use the simplifying assumption of 40 weeks gestation and six weeks postpartum for pregnancy state *intialization* in our cause model only. We could improve this assumption by using location-specific mean gestational age rather than 40 weeks as the assumed duration of pregnancy.

.. list-table:: State prevalence table
  :widths: 15 15 15
  :header-rows: 1

  * - Exposure category
    - Value
    - Note
  * - np
    - :math:`1 - (ASFR + ASFR * SBR) * 46 / 52`
    - 
  * - p
    - :math:`(ASFR + ASFR * SBR) * 40 / 52`
    - 

>   * - np

Reviewer 2: Do you mean pp here?

Author: Yes, updated thank you!

>   * - np
>     - :math:`1 - (ASFR + ASFR * SBR) * 6 / 52`

Reviewer: I think this might be a typo?

```suggestion
  * - pp
    - :math:`(ASFR + ASFR * SBR) * 6 / 52`
```

Author: Yes, this should be as you suggested. Thank you!!

    -  

.. list-table:: State transition data
  :header-rows: 1

  * - Source state
    - Sink state  
    - Transition rate
    - Note
  * - np
    - p
    - :math:`\frac{ASFR + ASFR * SBR}{prevalence_\text{np}}`
    - 
  * - p
    - pp
    - Informed by gestational age (see below section)
    - Duration-based transition
  * - np

>     - n

Reviewer 2: What is the n sink state here? I don't see it defined anywhere

Author: This was a typo and is now updated :) thanks

>   * - np
>     - n

Reviewer: Again, either a typo or I just don't understand this

```suggestion
  * - pp
    - np
```

Author: Yes, another typo! Thank you for your careful review

    - 6 weeks (42 days) duration
    - Duration-based transition

.. list-table:: Data values
  :header-rows: 1

  * - Parameter
    - Data type  
    - Data ID
    - Source
    - Note
  * - :math:`ASFR`
    - Covariate
    - 13
    - get_covariate_estimates
    - decomp_step='step4' or 'iterative' for GBD 2019, 'step3' or 'iterative' for GBD 2020
  * - :math:`SBR`
    - Covariate
    - 1106
    - get_covariate_estimates
    - decomp_step='step4' or 'iterative' for GBD 2019, 'step3' or 'iterative' for GBD 2020

.. note::

   A note on locations for the :ref:`IV Iron simulation <2019_concept_model_vivarium_iv_iron>`:

>       The ASFR covariate is available for Sub-Saharan Afric (location_id=166) and South Asia (location_id=159). The SBR covariate is not available for regional estimates.

Reviewer 2: Typo = *Africa

>       The ASFR covariate is available for Sub-Saharan Afric (location_id=166) and South Asia (location_id=159). The SBR covariate is not available for regional estimates.

Reviewer: ```suggestion
      The ASFR covariate is available for Sub-Saharan Africa (location_id=166) and South Asia (location_id=159). The SBR covariate is not available for regional estimates.
```

   We may restrict to ages 15 to 49 pending input from the BMGF.

Gestational age
~~~~~~~~~~~~~~~~~

Upon transition into the pregnancy state, each simulant should be assigned a gestational age according to the process described on the :ref:`low birthweight short gestation risk exposure document <2019_risk_exposure_lbwsg>`. This value will inform the duration that the simulant remains in the pregnancy state prior to transitioning to the postpartum state. Note that the gestational age distribution is measured in weeks and will need to be converted to the equivalent simulation time measure.

For simulants who are initialized into the pregnancy state at the start of the simulation:

   Assign the simulant a gestational age value and then sample a random value from a uniform distribution between zero and the assigned gestational age value. The randomly sampled value will represent the current gestational duration of that pregnancy. The simulant should remain in the pregnancy state prior to transitioning to the postpartum state for the duration equal to the assigned gestational age value *minus* the randomly sampled value.

.. note::

>    When we model maternal child dyads, the LBWSG exposure value assigned to the mother will be the exposure value assigned to the child in the dyad.

Reviewer 3: Awesome
